docs/content/topics/scripting/guide.md

```diff
@@ -9,1401 +9,867 @@ aliases:
+- the _event_ itself that this handler is for
 
-The image is a crucial part of the Brigade ecosystem. A container is created from this
-image, and it's within this container that we do the "real work". At the beginning, we
-explained that we think of Brigade scripts as "shell scripts for your cluster." When you
-execute a shell script, it is typically some glue code that manages calling one or more
-external programs in a specific way.
+The image is a crucial part of the Brigade ecosystem. A container is created
+from this image, and it's within this container that we do the "real work". At
+the beginning, we explained that we think of Brigade scripts as "shell scripts
+for your cluster." When you execute a shell script, it is typically some glue
+code that manages calling one or more external programs in a specific way.
 
-```
+```bash
 #!/bin/bash
 
 ps -ef "hello" | grep chrome
 ```
 
-The script above really just organizes the way we call two existing programs (`ps` and
-`grep`). Brigade does the same, except instead of executing _programs_, it executes
-_containers_. Each container is expressed as a Job, which is a wrapper that knows how to
-execute containers.
+The script above really just organizes the way we call two existing programs
+(`ps` and `grep`). Brigade does the same, except instead of executing
+_programs_, it executes _containers_. Each container is expressed as a Job,
+which is a wrapper that knows how to execute containers.
 
-So in our example above, we create a Job named "do-nothing" that runs an Alpine Linux
-container and (as the name implies) does nothing.
+So in our example above, we create a Job named "do-nothing" that runs a Debian
+Linux container and (as the name implies) does nothing.
 
-Jobs are created and run in different steps. This way, we can do some preparation on our
-job (as we will see in a moment) before executing it.
+Jobs are created and run in different steps. This way, we can do some
+preparation on our job (as we will see in a moment) before executing it.
 
-To run a job, we use the Job's `run()` method. Behind the scenes, Brigade will start a new
-`alpine:3.4` pod (downloading it if necessary), execute it, and monitor it. When the
-container is done executing, the run is complete.
+To run a job, we use the Job's `run()` method. Behind the scenes, Brigade will
+start a new `debian` pod (downloading it if necessary), execute it, and
+monitor it. When the container is done executing, the run is complete.
 
-It is worth noting that a `run()` method is _asynchronous_. That means that when we call
-`run()`, it will start processing in the background. Later on, we will see how we can take
-advantage of that to create pipelines.
+It is worth noting that a `run()` method is _asynchronous_. This means that
+when we call `run()`, it will start processing in the background. Later on, we
+will see how we can take advantage of that to create pipelines.
```

Comment:
Is it worth mentioning that `run()` is actually a request to the Brigade API server to _schedule_ the job for execution and that execution is subject to scheduling constraints (max concurrent jobs)? Is it worth further noting that `job.Run()` returns immediately after scheduling and the `await` keyword can (and usually should) be used to, instead, _block_ further script execution until the job has actually been executed and reached a terminal state?

Reply:
We should probably mention that `job.Run()` will throw an exception if the job fails.

We probably also need to mention (somewhere) that job success/failure is determined by its primary container's exit code.

We probably need to mention that same WRT the worker/script itself.